File: amazia-core/src/main/java/com/example/notification/controller/MarkConsoleNotificationReadController.java
package com.example.notification.controller;

import com.example.notification.service.MarkConsoleNotificationReadService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import org.springframework.web.server.ResponseStatusException;

import java.util.Map;

/**
 * フェーズ17 Step 6-0: Console 通知の既読化（設計書 §13.7.1）。
 * PUT /api/console/batch/notifications/{id}/read
 */
@RestController
@RequestMapping("/api/console/batch")
public class MarkConsoleNotificationReadController {

    private final MarkConsoleNotificationReadService service;

    public MarkConsoleNotificationReadController(MarkConsoleNotificationReadService service) {
        this.service = service;
    }

    @PutMapping("/notifications/{id}/read")
    public Map<String, Object> markRead(
            @PathVariable Long id,
            @RequestHeader(value = "X-User-Id", required = false) Long userId) {
        if (userId == null) {
            throw new ResponseStatusException(HttpStatus.UNAUTHORIZED, "X-User-Id required");
        }
        service.markRead(id, userId);
        return Map.of("id", id, "status", "read");
    }
}

File: amazia-console/app/Batch/Controller/MarkConsoleNotificationReadController.php
<?php

namespace App\Batch\Controller;

use App\Batch\Service\MarkConsoleNotificationReadService;
use App\Http\Controllers\Controller;
use Illuminate\Http\Request;

/**
 * フェーズ17 Step 6-2: 通知既読 Pass-through Controller。
 * PUT /api/console/batch/notifications/{id}/read
 */
class MarkConsoleNotificationReadController extends Controller
{
    private MarkConsoleNotificationReadService $service;

    public function __construct(MarkConsoleNotificationReadService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request, int $id)
    {
        $userId = $request->input('auth_user_id');
        $response = $this->service->markRead($userId, $id);
        return response()->json($response->json(), $response->status());
    }
}
